### ISSUE TITLE
Restyle buttons to look more modern

### ISSUE THREAD

**Issue by fitojb** [ubuntu/gtk-communitheme#211]
I simply can’t bear the flat gray chosen for the buttons. Can it be brightened up a little bit? It’ll help with #115.

**Comment by madsrh** [comment]
+1 @fitojb We know they look a bit old-school and we're investigating other possibilities. The current design uses [Ubuntu Touch as inspiration](https://community.ubuntu.com/t/mockups-new-design-discussions/1898/93).

I would love to hear/see if you have a suggestion!

**Comment by ya-d** [comment]
I don't think they're exactly depressing but I thought about that too. Two quick inspirations for mockups could be:
- The linked screenshot uses rgb-values of 253 for background and 245 for buttons while Communitheme uses 250 and 237. (Could be about color reproduction through image compression/rendering though.)
- I really like the new design for the check and radio buttons. Maybe analogously try white buttons? (I will propose something similar for slider switches soon, too.)
- Maybe scaling back the drop shadow a little (less dark) helps too. (Another issue to be filed: drop shadow for round buttons looks really weird.)

**Comment by ya-d** [comment]
Thanks for putting those together! 👍  So the more common theme seems to be brighter buttons (Mac OS too). Adwaita clearly (and much more so) suffers from the same muddy, triste look IMHO.

I like the United buttons (white with softer border and shadow) but I also like the current 3-dimensional inner shadow for active/pushed down state (though it might be a bit too soft/blured currently which makes it a bit unrealistic from a lighting perspective).

8 macOS (left is pushed down)
<img width="413" alt="macos buttons" src="https://user-images.githubusercontent.com/30781876/36718882-a09c0f6a-1ba3-11e8-97d9-245db362ed7a.png">

**Comment by NusiNusi** [comment]
IMHO a combination of 3 and 4 could fit the best. 
**shadows**: from 3 because they look the most realistic. 
**shape**: from 4 because the corners are more rounded and that would fit more to the suru-style.

**Comment by clobrano** [comment]
MacOS buttons work fine because the background is darker than communitheme's one. It's basically a color inversion and maybe also a gradient

**Comment by ya-d** [comment]
Regarding MacOS: Those were actually "header bar" buttons, sorry. In app buttons are straight white against rgb 236 or 227 background. Maybe @NusiNusi can replace the screenshot.

8 macOS buttons
<img width="366" alt="macos buttons 2" src="https://user-images.githubusercontent.com/30781876/36733231-3c8a8ca4-1bd0-11e8-9af5-5d8b376e6267.png">

**Comment by NusiNusi** [comment]
So what are the parts we need to think about?
- shadows/glow
- shape
- colour
- possible gradient
- size

**Comment by clobrano** [comment]
Also, macos button does not have any hover effect (they have very limited effects in general). Maybe simplifying is the way? :smile:

**Comment by madsrh** [comment]
Hey! Here's my humble opinion:

Adapta and POP buttons lacks contrast.
Adwaita, Adapta and POP buttons are HUGE.
Community theme, Adwaita and partially Ambiance looks dated.
Arc and Numix lacks borders or shadow and without either it reduce discoverability (is this a button?)

I really like Arc (and the derivatives) so below is a mockup based on that with a darker border and a shadow on buttons. I forgot the suru round corners to imagine these. _(click to view in full size)_

![buttons](https://user-images.githubusercontent.com/3986894/36763034-9298341a-1c26-11e8-812c-2f019642c329.gif)

**Comment by NusiNusi** [comment]
I like them! Especially the little shadow underneath the button is really nice! We already discussed this on another issue (which was closed #164 ), but: how would they react when hovered or clicked. This will surely depend on #184 . My opinion: your buttons should be implemented ASAP!

**Comment by clobrano** [comment]
Work in progress
![newbuttons](https://user-images.githubusercontent.com/2883614/36802879-64b1e696-1cb6-11e8-91b2-58d2519ebdba.gif)

with transition
![newbuttons-transition](https://user-images.githubusercontent.com/2883614/36802888-68189d98-1cb6-11e8-847a-123a855fb404.gif)

**Comment by madsrh** [comment]
@clobrano Wow, You're fast! The deactivated border looks darker than the active ones. Except for that, it looks really good!

**Comment by NusiNusi** [comment]
looks best with transition. With suru round corners like @madsrh planned it would look perfect. at least for me.

**Comment by madsrh** [comment]
@NusiNusi I acutally think this looks great

**Comment by clobrano** [comment]
Reopening this because gtk2 style hasn't not been addressed yet

**Comment by godlyranchdressing** [comment]
I've been working on buttons a little bit to fix the contrast and border issues. They're darkened _just_ a bit.

![buttons](https://user-images.githubusercontent.com/27529229/37842618-f82b909a-2e98-11e8-8dfb-2df0a49b1849.gif)

![flat](https://user-images.githubusercontent.com/27529229/37842617-f81379d8-2e98-11e8-915c-3ebf507bebb3.gif)

I was hoping to have a fusion of the old style and the new style. I'd really like the dark theme to have the bottom highlight like the old style as well, as I think the dark theme buttons look too flat but that depends on what everyone else thinks of it, of course.

The orange focus ring will need to be changed to `outline-offset: -3px;` or similar as you can see in the gif (for non-flat buttons) that it's hard to tell when a button is being hovered on while it's in focus (so I made hovering darken the background color a little).

**Comment by clobrano** [comment]
I like this proposal, just not sure about the dark border of pressed button. Maybe a little more similar to the active insensitive one?

**Comment by godlyranchdressing** [comment]
Updated: ![new](https://user-images.githubusercontent.com/27529229/37843792-511283d2-2e9c-11e8-8cbd-21f1f998bba3.jpg)

**Comment by godlyranchdressing** [comment]
![focus](https://user-images.githubusercontent.com/27529229/37843846-6d9ef26a-2e9c-11e8-9f99-51fdbbcaae79.gif)

**Comment by clobrano** [comment]
Nice!

**Comment by madsrh** [comment]
@godlyranchdressing:
> I've been working on buttons a little bit to fix the contrast and border issues. They're darkened just a bit.

I suggested (Hangout #150) a background change to fix the contrast issue. Did you (or @clobrano ) argue against the background change to #f7f7f7 (porcelain)?

**Comment by clobrano** [comment]
I'm not against changing bg color, in fact I wondered whether too much white could be annoying.

Here is a selection of intermediate color though, since I also lik

Current (#FDFDFD)
![image](https://user-images.githubusercontent.com/2883614/37863308-935d907c-2f5c-11e8-8904-0ba2c967f7b5.png)

porcelain (#F7F7F7)
![image](https://user-images.githubusercontent.com/2883614/37863379-8e75e82e-2f5d-11e8-9721-5a1d0c9173d1.png)

#F9F9F9
![image](https://user-images.githubusercontent.com/2883614/37863390-b7efd1ce-2f5d-11e8-8a3d-a1ab446763d2.png)

#FAFAFA
![image](https://user-images.githubusercontent.com/2883614/37863371-6aa06514-2f5d-11e8-98c0-f48769be2717.png)

**Comment by madsrh** [comment]
I guess we should move this conversation to issue #259 😄 

I'm still +1 for porcelain (#F7F7F7)

**Comment by clobrano** [comment]
No please :smile: , that was only sidebar's color

**Comment by NusiNusi** [comment]
IMHO the buttons looked this way the best:
https://user-images.githubusercontent.com/2883614/36802888-68189d98-1cb6-11e8-847a-123a855fb404.gif
what where the reasons for the change?
